src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -280,27 +429,72 @@ private function evaluateBracket(string $expr, mixed $value): array
 
             $result = [];
 
+            $allStringKeys = true;
+            foreach ($parts as $part) {
+                $part = trim($part);
+                if (!preg_match('/^([\'"])(.*)\1$/', $part)) {
+                    $allStringKeys = false;
+
+                    break;
+                }
+            }
+
+            if ($allStringKeys) {
+                if (!\is_array($value) || !array_is_list($value)) {
+                    foreach ($parts as $part) {
+                        $part = trim($part);
+
+                        if (!preg_match('/^([\'"])(.*)\1$/', $part, $matches)) {
+                            continue;
+                        }
+
+                        $key = JsonPathUtils::unescapeString($matches[2], $matches[1]);
+                        $result = array_merge($result, $this->getValueIfKeyExists($value, $key));
+                    }
+
+                    return $result;
+                }
+
+                foreach ($value as $item) {
+                    if (!\is_array($item)) {
+                        continue;
+                    }
+
+                    foreach ($parts as $part) {
+                        $part = trim($part);
+                        if (!preg_match('/^([\'"])(.*)\1$/', $part, $matches)) {
+                            continue;
+                        }
+
+                        $key = JsonPathUtils::unescapeString($matches[2], $matches[1]);
+                        $result = array_merge($result, $this->getValueIfKeyExists($item, $key));
+                    }
+                }
+
+                return $result;
+            }
+
             foreach ($parts as $part) {
                 $part = trim($part);
 
                 if ('*' === $part) {
-                    $result = array_merge($result, array_values($value));
+                    $result = array_merge($result, array_values((array) $value));
                 } elseif (preg_match('/^(-?\d*+)\s*+:\s*+(-?\d*+)(?:\s*+:\s*+(-?\d++))?$/', $part, $matches)) {
                     // slice notation
                     $sliceResult = $this->evaluateBracket($part, $value);
                     $result = array_merge($result, $sliceResult);
                 } elseif (preg_match('/^([\'"])(.*)\1$/', $part, $matches)) {
                     $key = JsonPathUtils::unescapeString($matches[2], $matches[1]);
 
-                    if (array_is_list($value)) {
+                    if (\is_array($value) && array_is_list($value)) {
                         // for arrays, find ALL objects that contain this key
                         foreach ($value as $item) {
-                            if (\is_array($item) && \array_key_exists($key, $item)) {
+                            if ($this->isArrayOrObject($item) && $this->hasKey($item, $key)) {
```

Comment:
method hasKey can be removed after this change:
```suggestion
                            if ($this->getValueIfKeyExists($item, $key)) {
```